src/popup/base.js

```diff
@@ -96,74 +101,153 @@ export const elements = {
 
-  top() {
-    return this.stack[this.stack.length - 1];
-  }
+    top() {
+      return this.stack[this.stack.length - 1];
+    },
 
-  isEmpty() {
-    return this.stack.length === 0;
-  }
+    isEmpty() {
+      return this.stack.length === 0;
+    },
 
-  clear() {
-    this.stack = [];
-  }
+    clear() {
+      this.stack = [];
+    },
+  };
 }
 
-export function initGlobalObject() {
-  window.appObject = {}; // global object to store the application variables
-  window.appObject.inputText = '';
-  window.appObject.pageNo = 1;
-  window.appObject.bookmarksSectionIdx = 0;
-  window.appObject.enableMatureContent = false;
-
-  // List to hold  selected by the user from the drop down.
-  window.appObject.userSelectedSourcesList = [];
-
-  // List to hold user selected licenses
-  window.appObject.userSelectedLicensesList = [];
+/**
+ * @desc Returns the ids of all the checked checkboxes inside the passed DOM element
+ * @param {HTMLElement} checkboxesWrapper
+ * @return {HTMLElement[]}
+ */
+function getCheckedCheckboxes(checkboxesWrapper) {
+  const checkboxes = checkboxesWrapper.querySelectorAll('input[type=checkbox]');
 
-  // List to hold user selected use case
-  window.appObject.userSelectedUseCaseList = [];
+  const checkedCheckboxes = [];
+  checkboxes.forEach(checkbox => {
+    if (checkbox.checked) checkedCheckboxes.push(checkbox.id);
+  });
 
-  window.appObject.userSelectedImageTypeList = [];
-  window.appObject.userSelectedImageSizeList = [];
-  window.appObject.userSelectedFileTypeList = [];
-  window.appObject.userSelectedAspectRatioList = [];
-
-  // object to store latest sources as <source_name, display_name>
-  window.appObject.sourcesFromAPI = {};
-
-  // Search Storage
-  window.appObject.storeSearch = {};
-
-  // store the name of the current active section
-  window.appObject.activeSection = 'search';
-  window.appObject.activeSearchContext = 'normal'; // possible values -> normal, collection, tag
-  window.appObject.collectionName = '';
-  window.appObject.clickedImageTag = false;
-
-  window.appObject.collectionSectionFilled = false;
-  window.appObject.bookmarksEditViewEnabled = false;
+  return checkedCheckboxes;
+}
 
-  window.appObject.imageDetailStack = new Stack();
+/**
+ * @desc Factory Function: creates and returns an object which is used to manage the application state
+ * @return {Object}
+ */
+function createApplicationObject() {
+  return {
+    inputText: '',
+    pageNo: 1,
+    // holds the current active section of the
+    // extension: filter, search, collections, bookmarks
+    activeSection: 'search',
+    // holds the context in which the search will
+    // happen: default, collection(search by sources), image-tag(search by image-tag)
+    searchContext: 'default',
+
+    // data structures to hold active filters
+    useCaseFilters: [],
+    licenseFilters: [],
+    sourceFilters: [],
+    fileTypeFilters: [],
+    imageTypeFilters: [],
+    imageSizeFilters: [],
+    aspectRatioFilters: [],
```

Comment:
bit of a refactor, but i think nesting these under a `filters` key would be cleaner:

```js
    filters: {
      useCase: [],
      license: [],
      source: [],
      fileType: [],
      imageType: [],
      imageSize: [],
      aspectRatio: [],
    },
```

makes them simpler to reset, too: 

```js
  for (const filter of Object.keys(filters)) {
    filters[filter] = []
  }
```